frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -45,16 +60,40 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
     browser,
   );
 
+  // Run onTokenChange callback
+  // store token in session storage or remove it when undefined
+  const applyToken = (tr: TokenResponse | undefined) => {
```

Comment:
Rather than having this as a function we could have it in a useEffect, like:

```
React.useEffect(() => {
 ....
}, [JSON.stringify(tokenResponse)]);

Reply:
that's where it was originally. but for a reason that I did not find, that use effect was not always triggered when I expected it to because the auth wrapper was not rerendered every time the tokenResponse got updated by the refresh function.
scenarios in which that happened: 
* reload page, wait for token to refresh, next request fails
* trigger a request, wait for token to refresh, then reload

Reply:
I'll see if the stringify solves that problem